shift-shift.go

```diff
@@ -292,3 +307,11 @@ func listenEvents(
 		replyTo <- Message{Device: kbd, Events: events}
 	}
 }
+
+func pressedBefore(t *time.Timer, timeout int) (*time.Timer, bool) {
```

Comment:
What does this function do?

Reply:
The function looks strange because it's named like it's some kind of checker or maybe helper but it actually has a side effect - it resets the timer and it's not obvious why it calls t.Stop().

Personally, I just didn't get what this function does. Maybe some documentation could help.

Reply:
Strange indeed, see some improvements.